lib/UmdMainTemplatePlugin.js

```diff
@@ -163,12 +173,12 @@ class UmdMainTemplatePlugin {
 				"})(this, function(" + externalsArguments(externals) + ") {\nreturn ", "webpack/universalModuleDefinition"), source, ";\n})");
 		}.bind(this));
 		mainTemplate.plugin("global-hash-paths", function(paths) {
-			if(this.name) paths = paths.concat(this.name);
+			if(this.names.root) paths = paths.concat(this.names.root);
 			return paths;
 		}.bind(this));
 		mainTemplate.plugin("hash", function(hash) {
 			hash.update("umd");
-			hash.update(`${this.name}`);
+			hash.update(`${this.names.root}`);
```

Comment:
@sokra Again, is it okay if names are repeated here, or do we need to dedupe first?